src/MensaWidget/MWMealCleaner.class.st

```diff
@@ -0,0 +1,148 @@
+"
+A MWMealCleaner is a FIlter for Meal lists provided by the parsing after processing the JSON String.
+
+Instance Variables
+
+"
+Class {
+	#name : #MWMealCleaner,
+	#superclass : #Object,
+	#category : #MensaWidget,
+	#'squeak_changestamp' : 'Max 6/11/2024 20:46'
+}
+
+{
+	#category : #cleaning,
+	#'squeak_changestamp' : 'Max 6/12/2024 13:56'
+}
+MWMealCleaner class >> clean: aMealList [
+	
+	| newMealList |
+	
+	newMealList := OrderedCollection new.
+	aMealList isEmpty ifTrue: [^ newMealList add: self emptyMeal; 
+		yourself].
+	aMealList do: [:each | newMealList add: (self fixMeal: each)].
+	^ newMealList
+		
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Max 6/12/2024 13:50'
+}
+MWMealCleaner class >> emptyMeal [
+
+	^ MWMeal new id: nil;
+		name: 'Hier gibt es nichts zu sehen.' translated;
+		category: nil;
+		notes: nil;
+		prices: self emptyPrice
+]
+
+{
+	#category : #constants,
+	#'squeak_changestamp' : 'Max 6/12/2024 13:42'
+}
+MWMealCleaner class >> emptyPrice [
+
+	^ Json readFromString: '{"students":null,"employees":null,"pupils":null,"others":null}'
+]
+
+{
+	#category : #cleaning,
+	#'squeak_changestamp' : 'Max 6/10/2024 20:02'
+}
+MWMealCleaner class >> fixCategories: aMeal [
+
+	aMeal category ifNil: [^ nil].
+	(aMeal category beginsWith: 'Abend') ifTrue: [aMeal category: 'dinner'].
+	aMeal category = 'dinner' ifFalse: [aMeal category: 'lunch']
+	
+	
+	
+		
+]
+
+{
+	#category : #cleaning,
+	#'squeak_changestamp' : 'Max 6/12/2024 13:51'
+}
+MWMealCleaner class >> fixMeal: aMeal [
+
+	| meal |
+	
+	meal := aMeal.
+	meal isSaladBar ifTrue: [meal := self saladBar].
+	meal isHotCounter ifTrue: [meal := self hotCounter].
+	self fixPrices: meal;
+	 	fixCategories: meal.
+	^ meal
+		
+]
+
+{
+	#category : #cleaning,
+	#'squeak_changestamp' : 'Max 6/12/2024 13:46'
+}
+MWMealCleaner class >> fixPrices: aMeal [
+	
+	| price |
+	
+	aMeal prices ifNil: [aMeal prices: self emptyPrice. 
+		^ nil].
+	price := aMeal prices.
+	"aMeal prices class = MWPrice ifTrue: [^ nil].
+	price := (MWParser put: '[', aMeal prices asJsonString, ']' in: MWPrice) first."
```

Comment:
Is this comment intentional or a debugging leftover?

Reply:
done